test/gtest/binary_tensor_ops.cpp

```diff
@@ -0,0 +1,295 @@
+                                           (dstSuperTensor.desc.GetNumDims() - lens.size()),
+                                       dstSuperStrides.end());
+
+        dstDesc     = miopen::TensorDescriptor(miopen_type<DstType>{}, lens, dstStrides);
+        dstDataSize = dstDesc.GetElementSpace() + offsets[1];
+
+        ASSERT_GE(srcSuperTensor.desc.GetElementSpace(), dstDataSize);
+
+        ASSERT_GE(srcSuperTensor.desc.GetNumDims(), lens.size());
+
+        const std::vector<size_t>& srcSuperStrides = srcSuperTensor.desc.GetStrides();
+        std::vector<size_t> srcStrides(srcSuperStrides.begin() +
+                                           (srcSuperTensor.desc.GetNumDims() - lens.size()),
+                                       srcSuperStrides.end());
+
+        srcDesc     = miopen::TensorDescriptor(miopen_type<SrcType>{}, lens, srcStrides);
+        srcDataSize = srcDesc.GetElementSpace() + offsets[0];
+
+        ASSERT_GE(srcSuperTensor.desc.GetElementSpace(), srcDataSize);
+    }
+
+    void RunCast()
+    {
+        std::vector<DstType> dstSuperCpu(dstSuperTensor.begin(),
+                                         dstSuperTensor.begin() + dstDataSize);
+        std::vector<SrcType> srcSuperCpu(srcSuperTensor.begin(),
+                                         srcSuperTensor.begin() + srcDataSize);
+
+        const auto [srcOffset, dstOffset] = miopen::tien<2>(std::get<std::vector<int>>(GetParam()));
+        const auto alpha                  = std::get<float>(GetParam());
+        const auto clamp                  = std::get<bool>(GetParam());
+
+        auto&& handle     = get_handle();
+        auto dstSuper_dev = handle.Write(dstSuperCpu);
+        auto srcSuper_dev = handle.Write(srcSuperCpu);
+
+        miopen::CastTensor(handle,
+                           &alpha,
+                           clamp,
+                           srcDesc,
+                           srcSuper_dev.get(),
+                           dstDesc,
+                           dstSuper_dev.get(),
+                           srcOffset,
+                           dstOffset);
+
+        auto result = handle.Read<DstType>(dstSuper_dev, dstDataSize);
+
+        if(clamp)
+        {
+            operate_over_subtensor(
+                [alpha, clampVal = static_cast<float>(std::numeric_limits<DstType>::max())](
+                    auto& dst, auto src) {
+                    dst = std::min(static_cast<float>(src) * alpha, clampVal);
+                },
+                dstSuperCpu,
+                srcSuperCpu,
+                dstDesc,
+                srcDesc,
+                dstOffset,
+                srcOffset);
+        }
+        else
+        {
+            operate_over_subtensor(
+                [alpha](auto& dst, auto src) { dst = static_cast<float>(src) * alpha; },
+                dstSuperCpu,
+                srcSuperCpu,
+                dstDesc,
+                srcDesc,
+                dstOffset,
+                srcOffset);
+        }
```

Comment:
Looks like we gained coverage here.
I think the old tests only checked the clamp case.

Nice!